lapce-data/src/source_control.rs

```diff
@@ -21,11 +23,14 @@ pub struct SourceControlData {
     pub widget_id: WidgetId,
     pub split_id: WidgetId,
     pub split_direction: SplitDirection,
+    // List of VCS modified files & whether they should be included in the next commit
+    pub file_list: Vec<(PathBuf, bool)>,
     pub file_list_id: WidgetId,
     pub file_list_index: usize,
     pub editor_view_id: WidgetId,
     pub commit_button_id: WidgetId,
-    pub file_diffs: Vec<(FileDiff, bool)>,
+    // Files and folders that should be colored based on their VCS status
+    pub file_diffs: HashMap<PathBuf, FileDiff>,
```

Comment:
I'm wondering if a ```IndexMap``` can consolidate ```file_list``` and ```file_diffs``` here?

Reply:
Oh, thanks! I wasn't familiar w/ `IndexMap`. Yes it looks like it will work just fine. I've updated this locally and will push after I finish a few more clean up items. Thank you!

Reply:
I really like how this turned out! Thank you!